### PR TITLE
GRIDEDIT-1491 Added api function to retrieve node, edge and face-node data only

### PR DESCRIPTION
**Pull request by BillSenior**

### CODE REVIEW

File: libs/MeshKernelApi/src/MeshKernel.cpp [hunk 1]
```diff
@@ -782,6 +782,32 @@ namespace meshkernelapi
         return lastExitCode;
     }
 
+    MKERNEL_API int mkernel_mesh2d_get_data_reduced(int meshKernelId, Mesh2D& mesh2d)
```
Comment (lucacarniato) [inline libs/MeshKernelApi/src/MeshKernel.cpp:785]: should mkernel_mesh2d_get_data be substituted with mkernel_mesh2d_get_data_reduced? if this is possible we have one api less to maintain in the other layers

**Review by lucacarniato — APPROVED** [review]

File: docs/Doxyfile.in [hunk 2]
```diff
@@ -1118,15 +1110,15 @@ VERBATIM_HEADERS       = YES
 # generated with the -Duse_libclang=ON option for CMake.
 # The default value is: NO.
 
-CLANG_ASSISTED_PARSING = NO
+# CLANG_ASSISTED_PARSING = NO
```
Comment (andreasbuykx) [inline docs/Doxyfile.in:1113]: do the changes in this file belong to this PR?

File: libs/MeshKernelApi/include/MeshKernelApi/MeshKernel.hpp [hunk 3]
```diff
@@ -1182,6 +1182,16 @@ namespace meshkernelapi
...
 
+        /// @brief Gets the reduced Mesh2D data
+        ///
+        /// This function ought to be called after `mkernel_mesh2d_get_dimensions` has been called
+        /// and the pointers have been set to correctly sized memory.
+        /// @param[in]     meshKernelId The id of the mesh state
+        /// @param[in,out] mesh2d       The Mesh2D data
+        /// @returns Error code
+        /// @note Only node, edge, face-node data will be retrieved (face node data include the number of nodes per face mesh2d.nodes_per_face)
+        MKERNEL_API int mkernel_mesh2d_get_data_reduced(int meshKernelId, Mesh2D& mesh2d);
```
Comment (andreasbuykx) [inline libs/MeshKernelApi/include/MeshKernelApi/MeshKernel.hpp:1193]: Would be nice to have more descriptive names, e.g. 
mkernel_mesh2d_get_connectivity (this method) and 
mkernel_mesh2d_get_computed_positions (filling the edgeX, edgeY, faceX, faceY arrays).

The .._get_data can then call both (or be made obsolete).

**Review by andreasbuykx — APPROVED** [review]

File: libs/MeshKernelApi/src/MeshKernel.cpp (hunk 1 above)
Comment (lucacarniato) [reply libs/MeshKernelApi/src/MeshKernel.cpp:785]: I would only keep mkernel_mesh2d_get_data and mkernel_mesh2d_get_data_reduced. i tested mkernel_mesh2d_get_node_edge_data in grid editor and it does work

**Review by lucacarniato — APPROVED** [review]

File: docs/Doxyfile.in (hunk 2 above)
Comment (lucacarniato) [reply docs/Doxyfile.in:1113]: looking at master id does not look like. i think chanegs in Doxyfile.in should be reverted

Comment (BillSenior) [reply docs/Doxyfile.in]: removed all doxygen file changes

**Pull request merged**